core/jazz_is-slack-channel-available/index.js

```diff
@@ -0,0 +1,169 @@
+
+const errorHandlerModule = require("./components/error-handler.js");
+const responseObj = require("./components/response.js");
+const configModule = require("./components/config.js");
+const logger = require("./components/logger.js")();
+const request = require('request');
+
+module.exports.handler = (event, context, cb) => {
+
+  //Initializations
+  var errorHandler = errorHandlerModule();
+  var config = configModule(event);
+  logger.init(event, context);
+  logger.info(event);
+
+  var resObj = {
+    'is_available': false
+  };
+
+  try {
+    genericInputValidation(event)
+      .then((res) => requestToChannels(config, resObj, res))
+      .then((res) => {
+        cb(null, responseObj(res, event.query));
+      })
+      .catch(error => {
+        if (error.result === 'inputError') {
+          return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+        } else if (error.result === 'unauthorized') {
+          return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+        } else {
+          cb(JSON.stringify(errorHandler.throwInternalServerError("Unhandled error.")));
+        }
+      });
+  } catch (e) {
+    cb(JSON.stringify(errorHandler.throwInternalServerError("Unhandled error.")));
+  }
+};
+
+function genericInputValidation(event) {
+  logger.info("inside genericInputValidation");
+  return new Promise((resolve, reject) => {
+    if (!event || !event.method) {
+      reject({
+        result: "inputError",
+        message: "method cannot be empty"
+      });
+    }
+
+    if (event.method !== 'GET') {
+      reject({
+        result: "inputError",
+        message: "Unsupported method/request"
+      });
+    }
+
+    if (event.method === 'GET' && (!event.query || Object.keys(event.query).indexOf('slack_channel') === -1)) {
+      reject({
+        result: "inputError",
+        message: "Missing input parameter slack_channel"
+      });
+    }
+
+    if (event.method === 'GET' && !event.query.slack_channel) {
+      reject({
+        result: "inputError",
+        message: "slack_channel value can not be empty/undefined"
+      });
+    }
+
+    if (!event.principalId) {
+      reject({
+        result: "unauthorized",
+        message: "Unauthorized."
+      });
+    }
+
+    resolve(event.query.slack_channel);
+  });
+}
+
+function requestToChannels(config, resObj, channel_name) {
+  logger.info("Inside requestToChannels:" + channel_name);
+  return new Promise((resolve, reject) => {
+    var slack_token = "token=" + config.slack_channel_token;
+    //Pull endpoints from config file
+    var public_channel_url = config.public_channel_endpoint + slack_token;
+    var priv_channel_url = config.priv_channel_endpoint + slack_token;
+    var urlList = [];
+    urlList.push(get_response(public_channel_url, channel_name));
+    urlList.push(get_response(priv_channel_url, channel_name));
+    Promise.all(urlList)
+      .then(res => {
+        if (res.indexOf('true') !== -1) {
```

Comment:
i guess this is an array you can use includes

Reply:
using incldes